### PR TITLE
basis.dom.event: extension for touch events added

### PR DESCRIPTION
**Pull request by fateevv**

### CODE REVIEW

File: src/basis/dom/event.js [hunk 1]
```diff
@@ -70,32 +71,22 @@
...
-    'mouseup',
-    'mouseover',
-    'mousemove',
-    'mouseout',
-    'mouseenter',
-    'mouseleave'
-  ].concat(BROWSER_EVENTS.mousewheel);
+  var DEPRECATED_PROPERTY = /^(returnValue|keyLocation|layerX|layerY|webkitMovementX|webkitMovementY|keyIdentifier)$/;
+
+  var KEYBOARD_EVENT = /^(keyup|keydown|keypress)$/;
```
Comment (lahmatiy) [inline src/basis/dom/event.js]: Solution with array is good enough. Don't turn it to RegExp's

File: src/basis/dom/event.js [hunk 2]
```diff
@@ -165,6 +156,13 @@
           wheelDelta: wheelDelta(event)
         });
       }
+      else if (isTouchEvent(event))
+      {
+        extend(this, {
+          mouseX: touchX(event),
```
Comment (lahmatiy) [inline src/basis/dom/event.js]: I think we should add `pointerX`/`pointerY` for mouse/touch/pointer events

File: src/basis/dom/event.js [hunk 3]
```diff
@@ -70,32 +71,22 @@
...
-    'mouseover',
-    'mousemove',
-    'mouseout',
-    'mouseenter',
-    'mouseleave'
-  ].concat(BROWSER_EVENTS.mousewheel);
+  var DEPRECATED_PROPERTY = /^(returnValue|keyLocation|layerX|layerY|webkitMovementX|webkitMovementY|keyIdentifier)$/;
+
+  var KEYBOARD_EVENT = /^(keyup|keydown|keypress)$/;
+  var TOUCH_EVENT = /^(touchstart|touchmove|touchend|touchcancel)$/;
```
Comment (lahmatiy) [inline src/basis/dom/event.js]: How about [pointer events](https://www.w3.org/TR/pointerevents/)?

Comment (fateevv) [inline src/basis/dom/event.js]: What exactly do you mean here?

Comment (lahmatiy) [inline src/basis/dom/event.js]: I mean pointer events also should be processed like mouse/touch events

File: src/basis/dom/event.js (hunk 2 above)
Comment (fateevv) [inline src/basis/dom/event.js]: `mouseX`/`mouseY` are using in several modules like `dragdrop.js`, `slider.js`, etc as base for various calculations, that's why we extend for example instance of `TouchEvent` by these non-native properties to provide common interface.
Should we replace them by `pointerX`/`pointerY` everywhere?

Comment (lahmatiy) [inline src/basis/dom/event.js]: `mouseX`/`mouseY` should stay for pointer/touch events for backward capability. But `pointerX`/`pointerY` should be implemented as well. And will be great if we change `mouse*` to `pointer*`

File: src/basis/dom/event.js [hunk 4]
```diff
@@ -90,13 +99,46 @@
...
+    KEYBOARD: 0,
+    MOUSE: 1,
+    TOUCH: 2,
+    POINTER: 3
+  };
+
+  function inputType(event){
+    var type = event.type;
+
+    if (KEYBOARD_EVENTS.indexOf(type) != -1)
```
Comment (lahmatiy) [inline src/basis/dom/event.js]: It's better make a map `eventName: type` and use here

```js
var eventType = {
  mousedown: INPUT_TYPE.MOUSE,
  ...
};

if (eventType.hasOwnProperty(type))
  return eventType[type];
```

File: src/basis/dom/event.js [hunk 5]
```diff
@@ -90,13 +99,46 @@
...
+    'pointerenter',
+    'pointerdown',
+    'pointermove',
+    'pointerup',
+    'pointercancel',
+    'pointerout',
+    'pointerleave'
+  ];
+
+  var INPUT_TYPE = {
```
Comment (lahmatiy) [inline src/basis/dom/event.js]: No need to use map of constants, better use just constants.

```js
var INPUT_TYPE_KEYBOARD = 0;
...
```

File: src/basis/dom/event.js [hunk 6]
```diff
@@ -70,33 +71,54 @@
...
+    pointerup:     TYPE_POINTER,
+    pointercancel: TYPE_POINTER,
+    pointerout:    TYPE_POINTER,
+    pointerleave:  TYPE_POINTER
+  };
 
-  function isMouseEvent(event) {
-    return MOUSE_EVENTS.indexOf(event.type) != -1;
-  }
+  extend(INPUT_TYPE, BROWSER_EVENTS.mousewheel.reduce(function(result, eventName){
```
Comment (lahmatiy) [inline src/basis/dom/event.js]: Looks a little bit complex. Maybe add those events explicit to INPUT_TYPE (with comment that those events are all about mouse wheel)?

**Pull request merged**